scipy/stats/_distn_infrastructure.py

```diff
@@ -1496,6 +1496,50 @@ def support(self, *args, **kwargs):
         place(out_b, 1-cond, self.badvalue)
         return out_a, out_b
 
+    def nnlf(self, theta, x):
```

Comment:
`nnlf`, `_nnlf`, `_nnlf_and_penalty`, and `_penalized_nnlf` have just been moved from `rv_continuous` to here. The only change is `_logpdf` is replaced with `_logpxf`, a new alias for `_logpdf`/`_logpmf` depending on whether the distribution is continuous or discrete.